prediction_market_agent/agents/full_set_collective_arbitrage_agent/deploy.py

```diff
@@ -0,0 +1,421 @@
+import math
+import typing as t
+from enum import Enum
+
+from prediction_market_agent_tooling.config import APIKeys
+from prediction_market_agent_tooling.deploy.agent import DeployableTraderAgent
+from prediction_market_agent_tooling.gtypes import (
+    CollateralToken,
+    OutcomeStr,
+    OutcomeToken,
+    Probability,
+)
+from prediction_market_agent_tooling.loggers import logger
+from prediction_market_agent_tooling.markets.agent_market import (
+    AgentMarket,
+    ProcessedTradedMarket,
+    SortBy,
+)
+from prediction_market_agent_tooling.markets.data_models import (
+    USD,
+    CategoricalProbabilisticAnswer,
+    PlacedTrade,
+    TradeType,
+)
+from prediction_market_agent_tooling.markets.market_fees import MarketFees
+from prediction_market_agent_tooling.markets.markets import MarketType
+from prediction_market_agent_tooling.markets.seer.seer import SeerAgentMarket
+from prediction_market_agent_tooling.markets.seer.seer_contracts import (
+    SwaprRouterContract,
+)
+from prediction_market_agent_tooling.tools.tokens.auto_deposit import (
+    mint_full_set_for_market,
+)
+from web3 import Web3
+
+
+class ArbitrageType(Enum):
+    UNDER_ESTIMATED = "underestimated"
+    OVER_ESTIMATED = "overestimated"
+
+
+class CalculationType(Enum):
+    COST = "cost"
+    REVENUE = "revenue"
+
+
+class DeployableFullSetCollectiveArbitrageAgent(DeployableTraderAgent):
+    supported_markets = [MarketType.SEER]
+
+    def load(self) -> None:
+        super().load()
+        self.epsilon = 0.02  # 2% error margin for probabilities as rounding error
+        self.swapr_router = SwaprRouterContract()
+        self.api_keys = APIKeys()
+        self.max_mint_amount_dollars: float = (
+            0.5  # Minting is done per 0.5$ outcome set
+        )
+        self.max_merge_amount_dollars: float = (
+            0.5  # Buying is done per 0.5$ outcome set
+        )
+
+    def run(self, market_type: MarketType) -> None:
+        if market_type not in self.supported_markets:
+            raise RuntimeError(
+                f"Can arbitrage only on {self.supported_markets} markets."
+            )
+        super().run(market_type=market_type)
+
+    def get_markets(
+        self,
+        market_type: MarketType,
+    ) -> t.Sequence[SeerAgentMarket]:
+        # Fetch all markets to choose from
+        available_markets = SeerAgentMarket.get_markets(
+            limit=5,
+            # sort_by=self.get_markets_sort_by,
+            sort_by=SortBy.HIGHEST_LIQUIDITY,
+            filter_by=self.get_markets_filter_by,
+            created_after=self.trade_on_markets_created_after,
+        )
+        return available_markets
```

Comment:
Does this need to be overriden? I would've expected the [logic](https://github.com/gnosis/prediction-market-agent-tooling/blob/6897bb44a8a8fffe35c564302fc6b0e9bd0afae0/prediction_market_agent_tooling/markets/seer/seer.py#L449) from `SeerAgentMarket` to be enough